### PR TITLE
[Wave8] Create a Workspace Card

### PR DESCRIPTION
**Pull request by MaciejSWM**
### Details

This PR introduces new Ideal Nav component: `WorkspaceCardCreateAWorkspace.tsx`.

It's just the component for further use in Ideal Nav project. To render it somewhere, add the code below to `SidebarLinks.js`:
```
// Import
import PressableWithFeedback from '@components/Pressable/PressableWithFeedback';
import Text from '@components/Text';
import WorkspaceCardCreateAWorkspace from '@pages/workspace/card/WorkspaceCardCreateAWorkspace';
// Import

            <View style={[styles.mh4, styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter, styles.mb3]}>
                <Text
                    style={styles.label}
                    color={theme.textSupporting}
                >
                    Workspace
                </Text>
                <PressableWithFeedback accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}>
                    {({hovered}) => (
                        <Icon
                            src={Expensicons.Plus}
                            width={12}
                            height={12}
additionalStyles={[styles.buttonDefaultBG, styles.borderRadiusNormal, styles.p2, hovered && styles.buttonHoveredBG]}
                        />
                    )}
                </PressableWithFeedback>
            </View>
            <WorkspaceCardCreateAWorkspace />

// themes.js update so that the sidebar BG is consistent with Figma
+ change `sidebar: colors.productDark200,` to `sidebar: sidebar: colors.productDark100,`
+ change `sidebar: colors.productLight200,`, to `sidebar: colors.productLight100,`
// themes.js update
```

image
image

### Fixed Issues

$ https://github.com/Expensify/App/issues/31766
PROPOSAL: 

### Tests

- Verify that no errors appear in the JS console
- `Plus` icon and the `Get Started` button should have Touchable effects on them. On press behaviour is not part of this PR though - clicking them won't do anything so far.

### Offline tests

- Same as `Tests`

### QA Steps

- Verify that no errors appear in the JS console
- Check the margins and paddings and sizes against [Figma designs](https://www.figma.com/file/Xv3yHJooZcxRGkXbAcTZD5/%23ideal-nav-(Ideal-Nav)?type=design&node-id=533-208590&mode=design&t=F2OL5065ZRb2ja3l-4).
- Click on `Plus` icon and the `Get Started` buttons - validate that Touchable effects are visible.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2023-12-13 at 13 08 17

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2023-12-13 at 12 34 03

</details>

<details>
<summary>iOS: Native</summary>

Screenshot 2023-12-13 at 12 29 27

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2023-12-13 at 12 30 10

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2023-12-13 at 12 31 07

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2023-12-13 at 12 27 10

</details>

### CODE REVIEW

**Comment by MaciejSWM** [comment]
Conclusions from slack discussion: For now two variants are OK - so `iconPosition` should stay. In the future cards will be unified to only have a single style.

**Comment by MaciejSWM** [comment]
Spanish translations slack thread: https://expensify.slack.com/archives/C01GTK53T8Q/p1701709855624649

**Comment by MaciejSWM** [comment]
@dannymcclain @shawnborton ready for your design check

**Comment by shawnborton** [comment]
For the small button here:
CleanShot 2023-12-05 at 08 26 23@2x


Can we confirm that it is using our standard button styles? It seems like the background color is incorrect. Also to confirm, it should be 28x28.

**Comment by shawnborton** [comment]
@dannymcclain @dubielzyk-expensify does the space between the headline and the text look right to you?
CleanShot 2023-12-05 at 08 27 09@2x

I think I was imagining those would be closer. Here is what we had in Figma:
image

**Comment by shawnborton** [comment]
I think that means we want a "normal" sized button in that card too, which is 40px tall

**Comment by dubielzyk-expensify** [comment]
Looks good to me!

**Comment by MaciejSWM** [comment]
@shawnborton thanks, I adjusted and double checked everything you mentioned. Result:
image

**Comment by MaciejSWM** [comment]
@aimane-chnaif ready for review

**Comment by shawnborton** [comment]
Looks good on my end.

**Review by BartoszGrajdek — APPROVED** [review]

**Comment by MaciejSWM** [comment]
image
image

**Review by aimane-chnaif — APPROVED** [review]
LGTM 🎉

File: src/components/Section/IconSection.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {View} from 'react-native';
+import Icon from '@components/Icon';
+import useThemeStyles from '@hooks/useThemeStyles';
```
Comment (mountiny) [inline src/components/Section/IconSection.js:5]: Does this have some dependency which is not migrated to TS or can we make this in TS now?

**Review by mountiny — APPROVED** [review]
thanks, seems like @shawnborton has agreed with the design here too so just one question

**Comment by kavimuru** [comment]
@MaciejSWM could you help with QA steps?  

> 1. Check the margins and paddings and sizes against [Figma designs](https://www.figma.com/file/Xv3yHJooZcxRGkXbAcTZD5/%23ideal-nav-(Ideal-Nav)?type=design&node-id=533-208590&mode=design&t=F2OL5065ZRb2ja3l-4).
> 2. Click on Plus icon and the Get Started buttons - validate that Touchable effects are visible.

We don't have access to Figma designs. Also which plus icon to click?

**Comment by mountiny** [comment]
@kavimuru you can chrck it off we will qa this in future prs once used

File: src/components/Section/IconSection.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const iconSectionPropTypes = {
+    icon: PropTypes.icon,
+    IconComponent: PropTypes.IconComponent,
+    iconContainerStyles: PropTypes.iconContainerStyles,
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/components/Section/IconSection.js:10]: Bumped that issue

[the remark this answers is not in the document]
Comment (blazejkustra) [reply src/components/Section/IconSection.js:10]: I fixed that [here](https://github.com/Expensify/App/pull/33645#discussion_r1436234932) @aimane-chnaif, funny enough you where chosen for C+ review 😸

**Pull request merged**